app/src/main/java/com/example/concordia_campus_guide/Helper/IndoorPathHeuristic.java

```diff
@@ -12,23 +12,35 @@
 
     Context context;
 
-    public IndoorPathHeuristic(Context context){
+    public IndoorPathHeuristic(Context context) {
         this.context = context;
     }
 
     protected double computeHeuristic(final WalkingPoint currentCoordinate, final WalkingPoint destinationPoint) {
+
         if (!currentCoordinate.getFloorCode().equalsIgnoreCase(destinationPoint.getFloorCode())) {
-            final WalkingPoint accessPoint = getNearestAccessPointForFloor(currentCoordinate);
-            return getEuclideanDistance(currentCoordinate, accessPoint)
-                    + getEuclideanDistance(accessPoint, destinationPoint);
+            String currentBuildingCode = currentCoordinate.getFloorCode().split("-")[0];
+            String destinationBuildingCode = destinationPoint.getFloorCode().split("-")[0];
```

Comment:
buildings don't have floor code. Thus, this might crash in that case. Did you test this case ?

Reply:
Good catch! I will look into it.

Reply:
After looking into it, you see that the currentCoordinate and desitinationPoint are actually WalkingPoints. Every walkingPoint has a floorcode so this is not an issue :)